### PR TITLE
Add fPIC option and forgotten ifdef conditionals

### PR DESCRIPTION
**Pull request by franksinankaya**

### CODE REVIEW

**Comment by franksinankaya** [comment]
@googlebot I signed it!

File: CMakeLists.txt [hunk 1]
```diff
@@ -79,6 +79,9 @@ option(ABSL_USE_GOOGLETEST_HEAD
 
 option(ABSL_RUN_TESTS "If ON, Abseil tests will be run." OFF)
 
+# link fails on arm64 and x86-64 without this
+add_compile_options(-fPIC)
```
Comment (derekmauro) [inline CMakeLists.txt:83]: Can you let us know what error you are seeing and how to reproduce it? This works for us.

**Review by derekmauro — CHANGES_REQUESTED** [review]
Please make this a pull request to the `master` branch instead.

File: CMakeLists.txt (hunk 1 above)
Comment (franksinankaya) [reply CMakeLists.txt:83]: hi, the yocto recipe for abseil-cpp has been merged here.

https://github.com/openembedded/meta-openembedded/tree/master/meta-oe/recipes-devtools/abseil-cpp

if you are familiar with yocto, you can do a build without the line:

https://github.com/openembedded/meta-openembedded/blob/696408ce3661b663a7895f75a01613e6cfeb8186/meta-oe/recipes-devtools/abseil-cpp/abseil-cpp_git.bb#L15

Comment (derekmauro) [reply CMakeLists.txt:83]: According to https://stackoverflow.com/a/38297422, `set(CMAKE_POSITION_INDEPENDENT_CODE ON)` is the idiomatic way of doing this.

Please send this as a PR to the `master` branch, thanks.

**Pull request closed without merging**